datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/search/AutoCompleteResolver.java

```diff
@@ -0,0 +1,67 @@
+package com.linkedin.datahub.graphql.resolvers.search;
+
+import com.linkedin.datahub.graphql.types.SearchableEntityType;
+import com.linkedin.datahub.graphql.exception.ValidationException;
+import com.linkedin.datahub.graphql.generated.AutoCompleteInput;
+import com.linkedin.datahub.graphql.generated.AutoCompleteResults;
+import com.linkedin.datahub.graphql.generated.EntityType;
+import com.linkedin.datahub.graphql.resolvers.ResolverUtils;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static com.linkedin.datahub.graphql.resolvers.ResolverUtils.bindArgument;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+/**
+ * Resolver responsible for resolving the 'autocomplete' field of the Query type
+ */
+public class AutoCompleteResolver implements DataFetcher<CompletableFuture<AutoCompleteResults>> {
+
+    private static final int DEFAULT_LIMIT = 5;
```

Comment:
above you had a field DEFAULT_COUNT which seemed to indicate the same thing, how are these two constants different?

Reply:
Only difference is there isn't full fledged paging in autocomplete. Instead, we just say "the max we can return". Think I should rename to count? Feel thats more consistent

Reply:
On second thought, I think we should keep as limit. This is a different feature from paging, as we are saying at "at most give me x results for the autocomplete". Resolving